src/overlays/actors/ovl_Door_Spiral/z_door_spiral.c

```diff
@@ -31,30 +40,223 @@ const ActorInit Door_Spiral_InitVars = {
+
+    point.x = player->actor.world.pos.x;
+    point.y = player->actor.world.pos.y + arg2;
+    point.z = player->actor.world.pos.z;
+
+    Actor_OffsetOfPointInActorCoords(&this->actor, &offset, &point);
+
+    if ((arg3 < fabsf(offset.x)) || (arg4 < fabsf(offset.y))) {
+        return FLT_MAX;
+    } else {
+        return offset.z;
+    }
+}
+
+s32 func_809A2EA0(DoorSpiral* this, PlayState* play) {
+    Player* player = GET_PLAYER(play);
+
+    if (Player_InCsMode(play) == 0) {
+        SpiralInfo* temp_v0 = &sSpiralInfoTable[this->unk148];
+        f32 temp_fv0 = func_809A2E08(play, this, 0.0f, temp_v0->unk_0B, temp_v0->unk_0C);
+
+        if (fabsf(temp_fv0) < 64.0f) {
+            s16 var_v0 = player->actor.shape.rot.y - this->actor.shape.rot.y;
+
+            if (temp_fv0 > 0.0f) {
+                var_v0 = 0x8000 - var_v0;
+            }
+            if (ABS_ALT(var_v0) < 0x3000) {
+                return (temp_fv0 >= 0.0f) ? 1.0f : -1.0f;
+            }
+        }
+    }
+    return 0;
+}
+
+void func_809A2FF8(DoorSpiral* this, PlayState* play) {
+    Player* player;
+    u32 index;
+
+    if (this->shouldClimb) {
+        DoorSpiral_SetupAction(this, func_809A3098);
+    } else if (func_809A2EA0(this, play) != 0) {
+        player = GET_PLAYER(play);
+        player->doorType = PLAYER_DOORTYPE_STAIRCASE;
+        player->doorDirection = this->unk146;
+        player->doorActor = &this->actor;
+        index = DOORSPIRAL_GET_FC00(&this->actor);
+        player->doorNext = (play->doorCtx.transitionActorList[index].params >> 0xA);
+        func_80122F28(player);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Door_Spiral/func_809A2B70.s")
+void func_809A3098(DoorSpiral* this, PlayState* play) {
+    Player* player = GET_PLAYER(play);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Door_Spiral/func_809A2BF8.s")
+    if (!(player->stateFlags1 & PLAYER_STATE1_20000000)) {
+        DoorSpiral_SetupAction(this, func_809A2DB0);
+        this->shouldClimb = false;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Door_Spiral/DoorSpiral_Init.s")
+void DoorSpiral_Update(Actor* thisx, PlayState* play) {
+    DoorSpiral* this = THIS;
+    Player* player = GET_PLAYER(play);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Door_Spiral/DoorSpiral_Destroy.s")
+    if (!(player->stateFlags1 & (PLAYER_STATE1_40 | PLAYER_STATE1_80 | PLAYER_STATE1_400 | PLAYER_STATE1_10000000)) ||
+        (this->actionFunc == func_809A2DB0)) {
+        this->actionFunc(this, play);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Door_Spiral/func_809A2DB0.s")
+void DoorSpiral_Draw(Actor* thisx, PlayState* play) {
+    s32 pad;
+    DoorSpiral* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Door_Spiral/func_809A2E08.s")
+    if (this->actor.objectSlot == this->objSlot) {
+        SpiralInfo* spiralInfo = &sSpiralInfoTable[this->unk148];
+        Gfx* temp = spiralInfo->unk_00[this->unk146];
```

Comment:
```suggestion
        Gfx* dList = spiralInfo->dLists[this->direction];
```
I think we worked out that this represents a staircase up or down. Could you add a comment to the struct? You could also update the name of the param getter `this->unk146 = DOORSPIRAL_GET_80(thisx);`